src/commands/org/create/sandbox.ts

```diff
@@ -155,9 +173,15 @@ export default class CreateSandbox extends SandboxCommandBase<SandboxCommandResp
       );
       delete sandboxReq.ActivationUserGroupName;
     }
+
     return {
       ...sandboxReq,
-      ...(this.flags.clone ? { SourceId: await this.getSourceId() } : {}),
+      ...(this.flags['source-sandbox-name'] ? { SourceId: await this.getSourceIdByName() } : {}),
+      ...(srcSandboxName
+        ? { SourceId: await requestFunctions.getSrcIdByName(this.flags['target-org'].getConnection(), srcSandboxName) }
+        : {}),
+      ...(srcId ? { SourceId: srcId } : {}),
+      ...(this.flags['source-id'] ? { SourceId: await this.getSourceId() } : {}),
```

Comment:
No need to check for flags and props, on L140 we already pass the source flags as `requestOptions` to `requestFunctions.createSandboxRequest`, `srcSandboxName` and `srcId` should have the flag/def. file values.